### DISCUSSION TITLE
Extract an estimate of the number of times a user "logs" in to Leaf from its logs

### DISCUSSION THREAD

**Discussion by artgoldberg** [uwrit/leaf#496] [Q&A] [answered]
Hi Folks

Our manage would like to obtain two metrics for the usage of Leaf:

- Cohorts created
- Logins

The first is easy -- look for log lines which satisfy `line['MessageTemplate'].startswith('FullCount cohort retrieved.')`.

But logins is tricky. Log lines that satisfy `line['MessageTemplate'] == "Authorization was successful."` occur much too frequently to represent "the user started using Leaf". It appears that they're generated by automated calls made by Leaf's GUI front-end. Is there another type of log line that would be better?

Thanks
Arthur

**Comment by ndobb** [comment, marked as answer]
Hi Arthur,

Our [leaf-log-parser repo](https://github.com/uwrit/leaf-log-parser) has a [handy script](https://github.com/uwrit/leaf-log-parser/blob/master/src/db/build.sql) where we define log records of interest for auditing and so on (which we create SQL views from).

["Created Access Token. Attestation:{@Attestation} Token:{Token}"](https://github.com/uwrit/leaf-log-parser/blob/master/src/db/build.sql#L508) is probably the log line to look for. That should contain the user, timestamp, and so on. Does that look like what you need?

Best,
-nic
